### PR TITLE
Fix for Store-1284 : User can edit the option-text and option field even they have readonly attribute true

### PR DESCRIPTION
**Pull request by tmkasun**
In this PR:

* Add `or` logic to check the page request more whether it is an edit call
* Update `renderOptionsForOptionsText` and `renderOptions` methods to accept `mode` parameter 
* Change the method invocation according to send `mode` parameter with them

Addresses the following issue: [STORE-1284](https://wso2.org/jira/browse/STORE-1284)

### CODE REVIEW

File: apps/publisher/themes/default/theme.js [hunk 1]
```diff
@@ -304,9 +304,9 @@ var engine = caramel.engine('handlebars', (function() {
             };
 
 
-            var renderOptionsForOptionsText = function(value, values, field) {
+            var renderOptionsForOptionsText = function(value, values, field,mode) {
```
Comment (daneshk) [inline apps/publisher/themes/default/theme.js]: formatting issue

File: apps/publisher/themes/default/theme.js [hunk 2]
```diff
@@ -304,9 +304,9 @@ var engine = caramel.engine('handlebars', (function() {
             };
 
 
-            var renderOptionsForOptionsText = function(value, values, field) {
+            var renderOptionsForOptionsText = function(value, values, field,mode) {
                 var id=field.name.tableQualifiedName+'_option';
-                var out = '<select ' + renderFieldMetaData(field,id) + '>';
+                var out = '<select ' + renderFieldMetaData(field,id,mode) + '>';
```
Comment (daneshk) [inline apps/publisher/themes/default/theme.js]: formatting issue

File: apps/publisher/themes/default/theme.js (hunk 1 above)
Comment (tmkasun) [inline apps/publisher/themes/default/theme.js]: Thanks, I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/842/commits/1b1eb4bc8b20375140097c9ce10e1c9bea4b97c0)

File: apps/publisher/themes/default/theme.js (hunk 2 above)
Comment (tmkasun) [inline apps/publisher/themes/default/theme.js]: Thanks, I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/842/commits/1b1eb4bc8b20375140097c9ce10e1c9bea4b97c0)

File: apps/publisher/themes/default/theme.js [hunk 3]
```diff
@@ -537,9 +543,9 @@ var engine = caramel.engine('handlebars', (function() {
                 var type = options.hash.type;
                 var tenantId = options.hash.tenantId;
                 var username = options.hash.username;
-                var isAuthorized =options.hash.auth ? options.hash.auth : false; 
+                var isAuthorized =options.hash.auth ? options.hash.auth : false;
```
Comment (splinter) [inline apps/publisher/themes/default/theme.js]: Formatting issue, please leave a space after =

File: apps/publisher/themes/default/theme.js [hunk 4]
```diff
@@ -560,9 +566,9 @@ var engine = caramel.engine('handlebars', (function() {
...
```
Comment (splinter) [inline apps/publisher/themes/default/theme.js]: Formatting issue, please leave a space after =

File: apps/publisher/themes/default/theme.js (hunk 3 above)
Comment (tmkasun) [inline apps/publisher/themes/default/theme.js]: Thanks @splinter , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/d132865cb7ce4b43f78012710e79d2772dab7797)

File: apps/publisher/themes/default/theme.js (hunk 4 above)
Comment (tmkasun) [inline apps/publisher/themes/default/theme.js]: Thanks @splinter , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/d132865cb7ce4b43f78012710e79d2772dab7797)

**Pull request merged**